source/fundamentals/linq.txt

```diff
@@ -0,0 +1,531 @@
+``Name`` field, with a secondary descending sort on their ``RestaurantId`` field.
+The following is a subset of the returned results:
+
+.. code-block:: json
+
+   // Results Truncated
+
+   ...
+   { "_id" : ObjectId(...), "name" : "Aba Turkish Restaurant", "restaurant_id" : "41548686", "cuisine" : "Turkish", "address" : {...}, "borough" : "Manhattan", "grades" : [...] }
+   { "_id" : ObjectId(...), "name" : "Abace Sushi", "restaurant_id" : "50006214", "cuisine" : "Japanese", "address" : { ... }, "borough" : "Manhattan", "grades" : [...] }
+   { "_id" : ObjectId(...), "name" : "Abacky Potluck", "restaurant_id" : "50011222", "cuisine" : "Asian", "address" : { ... }, "borough" : "Manhattan", "grades" : [...] }
+   { "_id" : ObjectId(...), "name" : "Abaleh", "restaurant_id" : "50009096", "cuisine" : "Mediterranean", "address" : { ... }, "borough" : "Manhattan", "grades" : [...] }
+   ...
+
+$lookup
+~~~~~~~
+
+The ``$lookup`` aggregation stage performs a left outer join to a collection in
+the same database to filter in documents from the "joined" collection for processing.
+The $lookup stage adds a new array field to each input document. The new array
+field contains the matching documents from the "joined" collection.
+
+.. note:: 
+
+   To perform a lookup you must make both collections queryable using the
+   ``AsQueryable`` method.
+
+   To learn how to make a collection queryable, see :ref:`csharp-linq-queryable`.
+
+Consider a second collection in the ``sample_restaurants`` database called
+``reviews`` that has restaurant reviews. The following ``Reviews`` class
+models the documents in that collection:
+
+.. literalinclude:: /includes/fundamentals/code-examples/linq.cs
+   :language: csharp
+   :dedent:
+   :start-after: start-review-model
+   :end-before: end-review-model
+
+Select the :guilabel:`Query Syntax` or :guilabel:`Method Syntax` tab to see how 
+how to generate an ``$lookup`` stage using LINQ:
+
+.. tabs::
+
+   .. tab:: Query Syntax
+      :tabid: query-syntax
+
+      .. code-block:: csharp
+
+         var query = from restaurant in queryableCollection
+             join rv in reviewCollection on restaurant.Name equals rv.RestaurantName into reviews
+             select new { restaurant, reviews };
+
+   .. tab:: Method Syntax
+      :tabid: method-syntax
+
+      .. code-block:: csharp
+
+         var query = queryableCollection
+             .GroupJoin(reviewCollection,
+                 restaurant => restaurant.Name,
+                 review => review.RestaurantName,
+                 (restaurant, reviews) =>
+                     new {restaurant, reviews}
+             );
+
+The preceding example returns documents from the ``restaurants`` collection, each
+with an added field called ``reviews``. The added field contains all documents from the
+``reviews`` collection with a ``name`` field that matches the ``name`` of the 
+document from the ``restaurants`` collection. 
```

Comment:
Issue: Could be a little clearer. This stuff is somehow both intuitive and tricky to articulate well

```suggestion
The preceding example returns all documents from the ``restaurants`` collection. Each restaurant document
has an added field called ``reviews``, which contains all reviews for that restaurant. 
A review matches a restaurant if the value of the ``name`` field in the review document matches the ``name`` field of the restaurant document. 
```